pkg/combustion/registry.go

```diff
@@ -227,22 +233,50 @@ func populateRegistry(ctx *image.Context, images []string) error {
 	arch := ctx.ImageDefinition.Image.Arch.Short()
 
 	for _, img := range images {
-		if err = storeImage(img, arch, logFile); err != nil {
-			return fmt.Errorf("adding image to registry store: %w", err)
-		}
-
 		convertedImage := strings.ReplaceAll(img, "/", "_")
 		convertedImageName := fmt.Sprintf("%s-%s", convertedImage, registryTarSuffix)
 
+		imageCacheLocation := filepath.Join(imageCacheDir, convertedImageName)
 		imageTarDest := filepath.Join(registryArtefactsPath(ctx), convertedImageName)
-		if err = generateRegistryTar(imageTarDest, logFile); err != nil {
-			return fmt.Errorf("generating registry store tarball: %w", err)
+
+		exists, err := imageExists(imageCacheLocation)
+		if err != nil {
+			return fmt.Errorf("checking if image already cached: %w", err)
 		}
 
+		if exists {
+			if err = fileio.CopyFile(imageCacheLocation, imageTarDest, fileio.ExecutablePerms); err != nil {
+				return fmt.Errorf("copying cached container image: %w", err)
+			}
+		} else {
+			if err = storeImage(img, arch, logFile); err != nil {
+				return fmt.Errorf("adding image to registry store: %w", err)
+			}
+
+			if err = generateRegistryTar(imageTarDest, logFile); err != nil {
+				return fmt.Errorf("generating registry store tarball: %w", err)
+			}
+
+			if err = fileio.CopyFile(imageTarDest, imageCacheLocation, fileio.ExecutablePerms); err != nil {
+				return fmt.Errorf("copying container image to cache: %w", err)
+			}
+		}
 		if err = bar.Add(1); err != nil {
 			zap.S().Debugf("Error incrementing the progress bar: %s", err)
 		}
 	}
 
 	return nil
 }
+
+func imageExists(imagePath string) (bool, error) {
```

Comment:
Nothing about this function is related to an image. It only checks whether a file exists. Perhaps it'd be a good addition to the `fileio` package?

Reply:
I think that'd be good